---
title: Spike-in quantitative abundance profiling reveals the microbial load variation of the root microbiome
author: "Qin Yuan"
date: "2019/3/31"
output: html_document
---

```{r warning=FALSE, message=FALSE}
rm(list=ls()) 

library(ggplot2)
library(ggpubr)
library(dplyr)
library(tidyr)
library(broom)
library(plyr)
library(scales)
library(pheatmap)
library("reshape2", quietly=T, warn.conflicts=F)
library(ggalluvial)
```


## Fig 2a

```{r}

spike <- read.csv("data/Fig2a-glm-group1.csv", header = T)

mod1 <- glm(counts ~ log2, family = poisson(link = "log"), data = spike)

p1 <- ggplot(spike, aes(log2, counts))+
  geom_smooth(method = "glm", size = 0.8, colour = "grey10") +
  geom_point(size = 2.5, colour = "grey20") +
  theme_bw() +
  scale_y_continuous(limits = c(0, 13000)) +
  labs(x="spike-in amount (log2 copies/reaction)", y="spike-in read counts") 

p1
# 
# ggsave(paste("result/Fig2a-glm-group1-20190218.pdf", sep=""), p1, width = 6, height = 6)


cor.test(spike$log2,spike$counts,method="pearson")

lm.fit1 <- lm(spike$counts~ spike$log2)
coef(lm.fit1)

```
## Fig 2b

```{r}
data1 <- read.csv("data/Fig2b-group1.csv", header = T)

data1G <- gather(data1, key = replicate , value = count, `sample01`:`sample05`)


p1G <- ggplot(data1G, aes(bacteria, count))+
  geom_boxplot(aes(fill = bacteria), na.rm = TRUE) +
  facet_wrap(~spike, ncol = 5) +
  scale_fill_manual(values = c("#ff8364", "#fdb87d", "#ffe8d5", "#fbd685","#7fa99b", 
                               "#d5eeff", "#87e0ff", "#53c7f0", "#1d97c1"))+
  labs(x="Bacteria", y="Count (%)")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0)) 
p1G
# ggsave(paste("result/Fig2b-group1.pdf", sep=""), p1G, width = 8, height = 6)

```



## Fig 2c - Fig 2d

```{r}
bac_ra <- read.csv("data/Fig2c-bar-ra.csv", header = T)
table_bac_ra_r <- gather(bac_ra,  key = sample , value = count, `sample01`:`sample04`)
table_bac_ra <- table_bac_ra_r
table_bac_ra$bacteria <- factor(table_bac_ra_r$bacteria)
table_bac_ra$group <- factor(table_bac_ra_r$group)
E0520_bac_ra <- filter(table_bac_ra, spike == "E0520")
group1_E0520_bac_ra <- filter(E0520_bac_ra , group == "1:01:01")
group2_E0520_bac_ra <- filter(E0520_bac_ra , group == "2:02:02")


bac_aa <- read.csv("data/Fig2d-bar-aa.csv", header = T)
table_bac_aa_r <- gather(bac_aa,  key = sample , value = count, `sample01`:`sample04`)
table_bac_aa <- table_bac_aa_r
table_bac_aa$bacteria <- factor(table_bac_aa_r$bacteria)
table_bac_aa$group <- factor(table_bac_aa_r$group)
E0520_bac_aa <- filter(table_bac_aa, spike == "E0520")
group1_E0520_bac_aa <- filter(E0520_bac_aa, group == "1:01:01")
group2_E0520_bac_aa <- filter(E0520_bac_aa, group == "2:02:02")

Fig2c_ra_g1g2 <- rbind(group1_E0520_bac_ra, group2_E0520_bac_ra)
Fig2d_aa_g1g2 <- rbind(group1_E0520_bac_aa, group2_E0520_bac_aa)

```


```{r}
Fig2c <- ggplot(Fig2c_ra_g1g2) +
  geom_boxplot(aes(x = bacteria, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#ffa07a"))+
  geom_line(aes(x = bacteria, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Bacteria", y="Relative abundance (%)")  +
  scale_x_discrete(limits=c("Act-101","Act-135","Act-140","Bac-186","Fir-11","Pro-1203","Pro-1204","Pro-670","Pro-672"))+
  scale_color_manual(values=c("#9acd32","#ffa07a"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig2c

#ggsave(paste("result/Fig2c-ra-g1g2-boxplot.pdf", sep=""), Fig2c, width = 7, height = 6)


Fig2d <- ggplot(Fig2d_aa_g1g2) +
  geom_boxplot(aes(x = bacteria, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#ffa07a"))+
  geom_line(aes(x = bacteria, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Bacteria", y="Quantitative abundance (relative to plant)")  +
  scale_x_discrete(limits=c("Act-101","Act-135","Act-140","Bac-186","Fir-11","Pro-1203","Pro-1204","Pro-670","Pro-672"))+
  scale_color_manual(values=c("#9acd32","#ffa07a"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig2d
#ggsave(paste("result/Fig2d-aa-g1g2-boxplot.pdf", sep=""), Fig2d, width = 7, height = 6)

```


## Fig 2e - Fig 2f

```{r}
bac_ra <- read.csv("data/Fig2e-bar-ra.csv", header = T)
table_bac_ra_r <- gather(bac_ra,  key = sample , value = count, `sample01`:`sample05`)
table_bac_ra <- table_bac_ra_r
table_bac_ra$bacteria <- factor(table_bac_ra_r$bacteria)
table_bac_ra$group <- factor(table_bac_ra_r$group)
E0520_bac_ra <- filter(table_bac_ra, spike == "E0520")
group1_E0520_bac_ra <- filter(E0520_bac_ra , group == "1:01:01")
group3_E0520_bac_ra <- filter(E0520_bac_ra , group == "2:02:01")

bac_aa <- read.csv("data/Fig2f-bar-aa.csv", header = T)
table_bac_aa_r <- gather(bac_aa,  key = sample , value = count, `sample01`:`sample05`)
table_bac_aa <- table_bac_aa_r
table_bac_aa$bacteria <- factor(table_bac_aa_r$bacteria)
table_bac_aa$group <- factor(table_bac_aa_r$group)
E0520_bac_aa <- filter(table_bac_aa, spike == "E0520")
group1_E0520_bac_aa <- filter(E0520_bac_aa, group == "1:01:01")
group3_E0520_bac_aa <- filter(E0520_bac_aa, group == "2:02:01")

Fig2e_ra_g1g3 <- rbind(group1_E0520_bac_ra, group3_E0520_bac_ra)
Fig2f_aa_g1g3 <- rbind(group1_E0520_bac_aa, group3_E0520_bac_aa)
```



```{r}
Fig2e <- ggplot(Fig2e_ra_g1g3) +
  geom_boxplot(aes(x = bacteria, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#87cefa"))+
  geom_line(aes(x = bacteria, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Bacteria", y="Relative abundance (%)")  +
  scale_x_discrete(limits=c("Act-101","Act-135","Act-140","Bac-186","Fir-11","Pro-1203","Pro-1204","Pro-670","Pro-672"))+
  scale_color_manual(values=c("#9acd32","#87cefa"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig2e
# ggsave(paste("result/Fig2e-ra-g1g3-boxplot.pdf", sep=""), Fig2e, width = 7, height = 6)


Fig2f <- ggplot(Fig2f_aa_g1g3) +
  geom_boxplot(aes(x = bacteria, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#87cefa"))+
  geom_line(aes(x = bacteria, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Bacteria", y="Quantitative abundance (relative to plant)")  +
  scale_x_discrete(limits=c("Act-101","Act-135","Act-140","Bac-186","Fir-11","Pro-1203","Pro-1204","Pro-670","Pro-672"))+
  scale_color_manual(values=c("#9acd32","#87cefa"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig2f
# ggsave(paste("result/Fig2f-aa-g1g3-boxplot.pdf", sep=""), Fig2f, width = 7, height = 6)

```

## Fig 2g

```{r}
## bac ra_aa error

spike <- read.csv("data/Fig2g-error.csv", header = T)
x <- c(0, 1, 1.3)
y <- c(0, 1, 1.3)
df <- data.frame(x=x, y=y)

bacplot <- ggplot(df, aes(x, y)) +
  geom_line() +
  geom_point(data = spike, aes(x = error_RA, y = error_AA, color = Spikeinamount), size = 2.5, na.rm = TRUE) + 
  theme_bw() +  
  scale_x_continuous(limits = c(0, 1.3)) + 
  scale_y_continuous(limits = c(0, 1.3)) + 
  labs(x="Bacteria RA error", y="Bacteria QA error")
bacplot

# ggsave(paste("result/Fig2g-error.pdf", sep=""), bacplot, width = 7, height = 6)

```

## Fig 3a

```{r}

spike <- read.csv("data/Fig3a-glm-group1.csv", header = T)

mod1 <- glm(counts ~ log2, family = poisson(link = "log"), data = spike)

p1 <- ggplot(spike, aes(log2, counts))+
  geom_smooth(method = "glm", size = 0.8, colour = "grey10") +
  geom_point(size = 2.5, colour = "grey20") +
  theme_bw() +
  scale_y_continuous(limits = c(0, 11000)) +
  labs(x="spike-in amount (log2 copies/reaction)", y="spike-in read counts") 

p1
# 
# ggsave(paste("result/Fig3a-glm-group1.pdf", sep=""), p1, width = 6, height = 6)


cor.test(spike$log2,spike$counts,method="pearson")

lm.fit1 <- lm(spike$counts~ spike$log2)
coef(lm.fit1)

```

## Fig 3b

```{r}
data1 <- read.csv("data/Fig3b-fun-nospike-group1.csv", header = T)


data1G <- gather(data1, key = replicate , value = count, `sample01`:`sample03`)


p1 <- ggplot(data1G, aes(fungi, count))+
  geom_boxplot(aes(fill = fungi), na.rm = TRUE) +
  facet_wrap(~spike, ncol = 5) +
  labs(x="Dilution gradient spike-in", y="Relative abundance (%)")+
  scale_fill_manual(values = c("#8ea6b4","#e7eff3", "#f8d5f0")) +
  scale_x_discrete(limits=c("Basi-AF78","Asco-AF1","Asco-AF105"))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0)) 

p1

# ggsave(paste("result/Fig3b-fun-nospike-group1.pdf", sep=""), p1, width = 8, height = 5)

```

## Fig 3c - Fig 3d

```{r}
fun_ra <- read.csv("data/Fig3c-bar-ra.csv", header = T)
table_fun_ra_r <- gather(fun_ra,  key = sample , value = count, `sample01`:`sample04`)
table_fun_ra <- table_fun_ra_r
table_fun_ra$fungi <- factor(table_fun_ra_r$fungi)
table_fun_ra$group <- factor(table_fun_ra_r$group)
E0520_fun_ra <- filter(table_fun_ra, spike == "E0520")
group1_E0520_fun_ra <- filter(E0520_fun_ra , group == "1:01:01")
group2_E0520_fun_ra <- filter(E0520_fun_ra , group == "2:02:02")


fun_aa <- read.csv("data/Fig3d-bar-aa.csv", header = T)
table_fun_aa_r <- gather(fun_aa,  key = sample , value = count, `sample01`:`sample04`)
table_fun_aa <- table_fun_aa_r
table_fun_aa$fungi <- factor(table_fun_aa_r$fungi)
table_fun_aa$group <- factor(table_fun_aa_r$group)
E0520_fun_aa <- filter(table_fun_aa, spike == "E0520")
group1_E0520_fun_aa <- filter(E0520_fun_aa, group == "1:01:01")
group2_E0520_fun_aa <- filter(E0520_fun_aa, group == "2:02:02")

Fig3c_ra_g1g2 <- rbind(group1_E0520_fun_ra, group2_E0520_fun_ra)
Fig3d_aa_g1g2 <- rbind(group1_E0520_fun_aa, group2_E0520_fun_aa)

```


```{r}
Fig3c <- ggplot(Fig3c_ra_g1g2) +
  geom_boxplot(aes(x = fungi, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#ffa07a"))+
  geom_line(aes(x = fungi, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Fungi", y="Relative abundance (%)")  +
  scale_x_discrete(limits=c("Basi-AF78","Asco-AF1","Asco-AF105"))+
  scale_color_manual(values=c("#9acd32","#ffa07a"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig3c
# ggsave(paste("result/Fig3c-ra-g1g2-boxplot.pdf", sep=""), Fig3c, width = 4, height = 6)



Fig3d <- ggplot(Fig3d_aa_g1g2) +
  geom_boxplot(aes(x = fungi, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#ffa07a"))+
  geom_line(aes(x = fungi, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Fungi", y="Quantitative abundance (relative to plant)")  +
  scale_x_discrete(limits=c("Basi-AF78","Asco-AF1","Asco-AF105"))+
  scale_color_manual(values=c("#9acd32","#ffa07a"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig3d
# ggsave(paste("result/Fig3d-aa-g1g2-boxplot.pdf", sep=""), Fig3d, width = 4, height = 6)

```

## Fig 3e - Fig 3f

```{r}
fun_ra <- read.csv("data/Fig3e-bar-ra.csv", header = T)
table_fun_ra_r <- gather(fun_ra,  key = sample , value = count, `sample01`:`sample04`)
table_fun_ra <- table_fun_ra_r
table_fun_ra$fungi <- factor(table_fun_ra_r$fungi)
table_fun_ra$group <- factor(table_fun_ra_r$group)
E0520_fun_ra <- filter(table_fun_ra, spike == "E0520")
group1_E0520_fun_ra <- filter(E0520_fun_ra , group == "1:01:01")
group3_E0520_fun_ra <- filter(E0520_fun_ra , group == "2:02:01")


fun_aa <- read.csv("data/Fig3f-bar-aa.csv", header = T)
table_fun_aa_r <- gather(fun_aa,  key = sample , value = count, `sample01`:`sample04`)
table_fun_aa <- table_fun_aa_r
table_fun_aa$fungi <- factor(table_fun_aa_r$fungi)
table_fun_aa$group <- factor(table_fun_aa_r$group)
E0520_fun_aa <- filter(table_fun_aa, spike == "E0520")
group1_E0520_fun_aa <- filter(E0520_fun_aa, group == "1:01:01")
group3_E0520_fun_aa <- filter(E0520_fun_aa, group == "2:02:01")

Fig3e_ra_g1g3 <- rbind(group1_E0520_fun_ra, group3_E0520_fun_ra)
Fig3f_aa_g1g3 <- rbind(group1_E0520_fun_aa, group3_E0520_fun_aa)

```


```{r}
Fig3e <- ggplot(Fig3e_ra_g1g3) +
  geom_boxplot(aes(x = fungi, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#87cefa"))+
  geom_line(aes(x = fungi, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Fungi", y="Relative abundance (%)")  +
  scale_x_discrete(limits=c("Basi-AF78","Asco-AF1","Asco-AF105"))+
  scale_color_manual(values=c("#9acd32","#87cefa"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig3e

Fig3f <- ggplot(Fig3f_aa_g1g3) +
  geom_boxplot(aes(x = fungi, y = count, fill = group), width = 0.3, na.rm = TRUE) +
  scale_fill_manual(values=c("#9acd32","#87cefa"))+
  geom_line(aes(x = fungi, y = mean, group = group, color = group), na.rm = TRUE) +
  labs(x="Fungi", y="Quantitative abundance (relative to plant)")  +
  scale_x_discrete(limits=c("Basi-AF78","Asco-AF1","Asco-AF105"))+
  scale_color_manual(values=c("#9acd32","#87cefa"))+
  theme_classic() +
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))
Fig3f

# ggsave(paste("result/Fig3e-ra-g1g3-boxplot.pdf", sep=""), Fig3e, width = 4, height = 6)
# ggsave(paste("result/Fig3f-aa-g1g3-boxplot.pdf", sep=""), Fig3f, width = 4, height = 6)

```

## Fig 3g

```{r}
## fun ra_aa error

spike2 <- read.csv("data/Fig3g-error.csv", header = T)
x <- c(0, 1, 1.1)
y <- c(0, 1, 1.1)
df <- data.frame(x=x, y=y)

funplot <- ggplot(df, aes(x, y)) +
  geom_line() +
  geom_point(data = spike2, aes(x = error_RA, y = error_AA, color = Spikeinamount),size = 2.5, na.rm = TRUE) + 
  theme_bw() +  
  scale_x_continuous(limits = c(0, 1.1)) + 
  scale_y_continuous(limits = c(0, 1.1)) + 
  labs(x="Fungi RA error", y="Fungi AA error") 
funplot
# ggsave(paste("result/Fig3g-error.pdf", sep=""), funplot, width = 7, height = 6)

```


# Fig 4a

```{r}
bac <- read.csv("data/Fig4a-lm-bac.csv", header = T)
fun <- read.csv("data/Fig4a-lm-fun.csv", header = T)

## Bacteria E05-E00
data_e05_bac <- dplyr::select(bac, OTUID, E05_bac, E00_bac)

x1 <- data_e05_bac$E05_bac
y1 <- data_e05_bac$E00_bac

cor.test(x1,y1,method="pearson")
lm.fit1 <- lm(y1~0+x1)
coef(lm.fit1)

plot(data_e05_bac$E05_bac,data_e05_bac$E00_bac,
     xlab ="OTU reads (%) Unamended root DNA", 
     ylab ="OTU reads (%) spiked root DNA")

abline(lm.fit1)
```

## Fig 4b

```{r}
## Fungi E05-E00
data_e05_fun <- dplyr::select(fun, OTUID, E05_fun, E00_fun)

x4 <- data_e05_fun$E05_fun
y4 <- data_e05_fun$E00_fun

cor.test(x4,y4,method="pearson")
lm.fit4 <- lm(y4~0+x4)
coef(lm.fit4)

plot(data_e05_fun$E05_fun,data_e05_fun$E00_fun,pch=2,
     xlab ="OTU reads (%) Unamended root DNA", 
     ylab ="OTU reads (%) spiked root DNA")

abline(lm.fit4)
```

## Fig 4c - Fig 4d

```{r}
bac <- read.csv("data/Fig4c-bac-all.csv")
fun <- read.csv("data/Fig4c-fun-all.csv")
fun2<- read.csv("data/Fig4c-fun-point.csv")


## E05-BAC-RA

t1 <- dplyr::select(bac, Bac, E05_RA_ONE, E05_RA_TWO)

bac05ra <- gather(t1, key = variable , value = count, `E05_RA_ONE`:`E05_RA_TWO`)

p1 = ggplot(bac05ra, aes(x=variable, y = count, fill = Bac )) + 
  geom_bar(stat = "identity",position="fill", width=0.7)+ 
  scale_fill_manual(values = c("#ff8364", "#fbd685", 
                               "#7fa99b", "#f6f6e9", "#1d97c1"))+
  scale_y_continuous(labels = scales::percent) + 
  xlab("Groups")+
  ylab("Relative Abundance (%)")+ 
  theme_classic()
p1

## E05-BAC-QA

t4 <- dplyr::select(bac, Bac, E05_QA_ONE, E05_QA_TWO)

bac05qa <- gather(t4, key = variable , value = count, `E05_QA_ONE`:`E05_QA_TWO`)

p4 = ggplot(bac05qa, aes(x=variable, y = count, fill = Bac )) + 
  geom_bar(stat = "identity", width=0.7) + 
  scale_fill_manual(values = c("#ff8364", "#fbd685", 
                               "#7fa99b", "#f6f6e9", "#1d97c1"))+
  xlab("Groups")+
  ylab("Quantitative Abundance (relative to host plant)")+ 
  theme_classic()
p4

```

## Fig 4e - Fig 4f

```{r}
## E05-FUN-RA

f1 <- dplyr::select(fun, FUN, E05_RA_ONE, E05_RA_TWO)

fun05ra <- gather(f1, key = variable , value = count, `E05_RA_ONE`:`E05_RA_TWO`)

fp1 = ggplot(fun05ra, aes(x=variable, y = count, fill = FUN )) + 
  geom_bar(stat = "identity",position="fill", width=0.7)+ 
  scale_fill_manual(values = c("#8ea6b4", "#f8d5f0","#89a4c7"))+
  scale_y_continuous(labels = scales::percent) + 
  xlab("Groups")+
  ylab("Relative Abundance (%)")+ 
  theme_classic()
fp1

## E05-FUN-QA

f3 <- dplyr::select(fun, FUN, E05_QA_ONE, E05_QA_TWO)

fun05qa <- gather(f3, key = variable , value = count, `E05_QA_ONE`:`E05_QA_TWO`)

fp3 = ggplot(fun05qa, aes(x=variable, y = count, fill = FUN )) + 
  geom_bar(stat = "identity", width=0.7) + 
  scale_fill_manual(values = c("#8ea6b4", "#f8d5f0","#89a4c7"))+
  xlab("Groups")+
  ylab("Quantitative Abundance (relative to host plant)")+ 
  theme_classic()
fp3

```

## Fig 5a

```{r}
load <- read.csv("data/Fig5f_sample.csv", header = T)

data_AH <- filter(load, Site == "AH")
data_HN <- filter(load, Site == "HN")
data_AH_BAC <- filter(data_AH, Micro == "bac")
data_AH_FUN <- filter(data_AH, Micro == "fungi")
data_AH_SUM <- filter(data_AH, Micro == "bacfun")

data_HN_BAC <- filter(data_HN, Micro == "bac")
data_HN_FUN <- filter(data_HN, Micro == "fungi")
data_HN_SUM <- filter(data_HN, Micro == "bacfun")
```




```{r}
load_mean <- read.csv("data/Fig5f.csv", header = T)

data_HN_mean <- filter(load_mean, Site == "HN")

p4 <- ggplot(data_HN_mean , aes(x= Type, y = Mean, fill = Micro )) + 
  geom_bar(stat = "identity", width=0.7) + 
  geom_errorbar(aes(ymin=Mean2-Error, ymax=Mean2+Error), width=.1) +
  labs(x="Group", y="Quantitative kingdom abundance (QA)") +
  facet_wrap(~Genotype, ncol = 2) +
  scale_fill_manual(values = c("#fdffab","#a8e6cf"))+
  theme_bw()
p4
```

## Fig 5b

```{r}
data5c = read.table("./data/Fig5c_new.txt",header = T, row.names = 1)
head(data5c)
data5c_stat  = read.table("./data/Fig5c_new_stat.txt",header = T, row.names = 1)
head(data5c_stat)
```

```{r}
com_5c_w <- compare_means( ratio ~ treatment, data = data5c, method = "wilcox.test", paired=FALSE,  group.by = "genotype")
com_5c_t <- compare_means( ratio ~ treatment, data = data5c, method = "t.test", paired=FALSE,  group.by = "genotype")
com_5c <- rbind(com_5c_w, com_5c_t)
head(com_5c)
```


```{r}
p = ggplot(data5c_stat, aes(x = treatment, y = radio_mean)) + 
  geom_bar(stat = "identity",fill = "white", colour = "black", width=0.7, na.rm = TRUE)+ 
  geom_errorbar(aes(ymin= radio_mean - radio_error, ymax= radio_mean + radio_error), width=.1) + 
  facet_wrap(~genotype)+
  xlab("Groups")+
  ylab("Ratio of bacteria to host plant")+
  theme_bw()
p

```

## Fig 5c

```{r}
corr <- read.table("data/Fig-0215-correlationplot.txt", header = T)

x1 <- corr$ratio
y1 <- corr$load

cor.test(x1,y1,method="pearson")
lm.fit1 <- lm(y1~0+x1)
coef(lm.fit1)

plot(corr$ratio,corr$load,
     xlab =" Ratios of microbial reads to host plant reads", 
     ylab ="Microbial load generated by Si-QAP")

abline(lm.fit1)


p <- ggplot(corr, aes(ratio, load)) +
  geom_point(size = 2.5) +
  theme_classic()+
  geom_smooth(method = "glm", size = 0.8, colour = "grey10")+
  labs(x="Ratios of microbial reads to host plant reads", y="Microbial load generated by Si-QAP") 

p
# 
# 
# ggsave(paste("result/Fig-0215-correlationplot.pdf", sep=""), p, width = 7, height = 6)

```

## Fig 5d

```{r}
data <- read.csv("data/Fig5c-heatmap-p.csv", header = T)

## HN

data_hn <- dplyr::select(data, Bacteria, starts_with("HN"))

all <- gather(data_hn, key = group , value = logFC, `HN_MH_RA`:`HN_WYJ_AA`)

p_hn <- ggplot(all, aes(group, Bacteria)) + 
  geom_tile(aes(fill = logFC),colour = "grey50") + 
  scale_fill_gradient2(low = "#6D9EC1",high = "sandybrown", midpoint = 0)+
  theme(axis.text.x = element_text(angle = -45, vjust = 1, hjust = 0))+
  scale_y_discrete(limits=c("Gammaproteobacteria","Deltaproteobacteria","Betaproteobacteria",
                            "Alphaproteobacteria","Verrucomicrobia","Spirochaetes","Nitrospirae",
                            "Ignavibacteriae","Firmicutes","Chloroflexi","Bacteroidetes","Actinobacteria",
                            "Acidobacteria"))+
  scale_x_discrete(limits=c("HN_MH_RA","HN_MH_AA","HN_WYJ_RA","HN_WYJ_AA"))

p_hn

```

## Fig 5f - Fig 5h
```{r}
data_ra <- read.csv("data/Fig5e-otu-bar-ra.csv", header = T)
data_aa <- read.csv("data/Fig5e-otu-bar-aa.csv", header = T)


## OTU16-RA
data2 <- filter(data_ra, OTUS == "OTU_16")

p2 <- ggplot(data2) +
  geom_boxplot(aes(x = Treatment, y = RA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = RA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_16", y="Relative genus abundance (RA) (‰)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p2

## OTU11-RA
data1 <- filter(data_ra, OTUS == "OTU_11")

p1 <- ggplot(data2) +
  geom_boxplot(aes(x = Treatment, y = RA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = RA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_11", y="Relative genus abundance (RA) (‰)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p1

## OTU13-RA
data3 <- filter(data_ra, OTUS == "OTU_13")

p3 <- ggplot(data3) +
  geom_boxplot(aes(x = Treatment, y = RA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = RA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_13", y="Relative genus abundance (RA) (‰)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p3

## OTU16-AA

data5 <- filter(data_aa, OTUS == "OTU_16")

p5 <- ggplot(data5) +
  geom_boxplot(aes(x = Treatment, y = AA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = AA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_16", y="Quantitative genus abundance (QA)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p5

## OTU16-AA

data4 <- filter(data_aa, OTUS == "OTU_11")

p4 <- ggplot(data4) +
  geom_boxplot(aes(x = Treatment, y = AA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = AA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_11", y="Quantitative genus abundance (QA)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p4

## OTU13-AA

data6 <- filter(data_aa, OTUS == "OTU_13")

p6 <- ggplot(data5) +
  geom_boxplot(aes(x = Treatment, y = AA), width = 0.3, na.rm = TRUE) +
  geom_jitter(aes(x = Treatment, y = AA, color = Treatment), width = 0.08, na.rm = TRUE) +
  labs(x="Group_OTU_13", y="Quantitative genus abundance (QA)") +
  scale_colour_manual(values = c("goldenrod1", "deepskyblue")) +
  theme_bw()
p6

```


## Fig 6a

## Fig 6b


## Fig 6c

```{r}
data = read.table("./data/cluster_total_load.txt",header = T, row.names = 1)
head(data)

sub_data = subset(data, SampleID %in% c("TAD2", "TAD3", "TAD4", "TAD6", "TAD7", "TAD9", "TAD10", "TAH1", "TAH3", "TAH4", "TAH5", "TAH6", "TAH8", "TAH10"))
```


```{r}
com_w1 <- compare_means( Totalload ~ Treatment, data = sub_data, method = "wilcox.test", paired=FALSE, group.by = "Micro")
com_t1 <- compare_means( Totalload ~ Treatment, data = sub_data, method = "t.test", paired=FALSE, group.by = "Micro")
head(com_t1)
head(com_w1)

com <- rbind(com_w1, com_t1)
# write.table(com , file = "./ttload_cluster_com.txt",sep = '\t',row.names = T)
```


```{r}
p1 = ggbarplot(sub_data, x = "Treatment", y = "Totalload", add = "mean_se", facet.by = "Micro",ylim = c(0, 2500))+
  stat_compare_means(method = "t.test", label =  "p.signif", label.y = 2500, label.x = 1.5)  
p1
# ggsave(paste("./ttload_cluster_com.pdf", sep=""), p1, width = 6, height = 8)
# ggsave(paste("./ttload_cluster_com.png", sep=""), p1, width = 6, height = 8)

```

## Fig 6d

```{r}
# Public file 1. "design.txt"  Design of experiment
design = read.table("./data/design.txt", header=T, row.names= 1, sep="\t") 

# Public file 2. "otu_table.txt"  raw reads count of each OTU in each sample
otu_table = read.delim("./data/otu_plant_aa.txt", row.names= 1,  header=T, sep="\t")

# Public file 3. "rep_seqs_tax.txt"  taxonomy for each OTU, tab seperated
taxonomy = read.delim("./data/taxonomy_8.txt", row.names= 1,header=F, sep="\t")
colnames(taxonomy) = c("kingdom","phylum","class","order","family","genus","species")

head(design)
head(otu_table)
head(taxonomy)

tax_count = merge(taxonomy, otu_table, by="row.names")

```

```{r}
## remove the abnormal points
sub_tax_count = dplyr::select(tax_count, Row.names, kingdom, phylum, class, order, family, genus, species, TAD2, TAD3, TAD4, TAD6, TAD7, TAD9, TAD10, TAH1, TAH3, TAH4, TAH5, TAH6, TAH8, TAH10)



g1_Bipolaris = dplyr::filter(sub_tax_count, genus == "Bipolaris")
g2_Fusarium = dplyr::filter(sub_tax_count, genus == "Fusarium")
genus = rbind(g1_Bipolaris, g2_Fusarium)
otuid = as.data.frame(genus$Row.names)
colnames(otuid) = "Row.names"


com_genus = merge(sub_tax_count, otuid, by="Row.names")
colnames(com_genus)[1] = "OTUID"


com_genus_tidy = gather(com_genus, key = SampleID , value = method2aa, `TAD2`:`TAH10`)


design2 = read.table("./data/design.txt", header=T, sep="\t") 
com_genus_tidy_design = merge(com_genus_tidy, design2, by="SampleID")
com_aa = dplyr::select(com_genus_tidy_design, SampleID, OTUID, genus, method2aa, Genotype)
colnames(com_aa)[3] = "Genus"
colnames(com_aa)[5] = "Treatment"
# 
# write.table(com_aa, file = "./pointD1D5D8H2H7H9/com_aa.txt",sep = '\t',row.names = T)
```



```{r}
## 读取com_ra数据
com_ra <-  read.table("./data/com_ra.txt", header=T, sep="\t") 
com_method1ra = gather(com_ra, key = Method , value = value, `method1ra`)
com_method2aa = gather(com_aa, key = Method , value = value, `method2aa`)

## 整合compare总表
com_aara <- rbind(com_method1ra, com_method2aa)
```



```{r}
## 使用两种方法检测OTU在不同样品中的丰度
OTU45 <- dplyr::filter(com_aara, OTUID == "OTU_45")

p45 <- ggboxplot(OTU45, x="Treatment",y = "value",
               facet.by = "Method", color = "Treatment", order = c("Disease","Healthy"),
               palette =  "uchicago", add = "jitter", short.panel.labs = FALSE, title = "OTU_45_Bipolaris")
# 添加p值
p045 <- p45 + stat_compare_means(label =  "p.signif",  method = "wilcox.test",label.x = 1.55)

p045

# ggsave(paste("./pointD1D5D8H2H7H9/com_aa_OTU_45.pdf", sep=""), p032, width = 6, height = 7)
# ggsave(paste("./pointD1D5D8H2H7H9/com_aa_OTU_45.png", sep=""), p032, width = 6, height = 7)
```


## Fig 6e

```{r}
data = read.table("./data/Fig6g.txt",header = T, row.names = 1)
head(data)

data_1 <- filter(data, re == "1")
data_2 <- filter(data, re == "2")
```

```{r}
com_1 <- compare_means( ratio ~ treatment, data = data_1, method = "wilcox.test", paired=FALSE)
com_t1 <- compare_means( ratio ~ treatment, data = data_1, method = "t.test", paired=FALSE)
com_2 <- compare_means( ratio ~ treatment, data = data_2, method = "wilcox.test", paired=FALSE)
com_t2 <- compare_means( ratio ~ treatment, data = data_2, method = "t.test", paired=FALSE)
head(com_1)
head(com_t1)
head(com_2)
head(com_t2)

com <- rbind(com_1, com_t1, com_2, com_t2)

# 
# write.table(com , file = "./result/Fig6_load_com.txt",sep = '\t',row.names = T)
```

```{r}
p1 = ggbarplot(data_1, x = "treatment", y = "ratio", add = "mean_se")+
  stat_compare_means(method = "t.test")  
p1
# ggsave(paste("./result/Fig6-load-1.pdf", sep=""), p1, width = 3, height = 6)
```



