---
title: "Figure6"
author: "Qin Yuan"
date: "2019/4/1"
output: html_document
---


```{r warning=FALSE, message=FALSE}
rm(list=ls()) 

library(ggplot2)
library(ggpubr)
library(dplyr)
library(tidyr)
library(broom)
library(plyr)
library(scales)
library(pheatmap)
library("reshape2", quietly=T, warn.conflicts=F)
library(ggalluvial)
```

## Fig 6a

```{r}
# Compare each group distance matrix by vegan adonis in bray_curtis
da_adonis = function(sampleV){
  sampleA = as.matrix(sampleV$sampA)
  sampleB = as.matrix(sampleV$sampB)
  design2 = subset(sub_design, group %in% c(sampleA,sampleB))
  if (length(unique(design2$group))>1) {
    sub_dis_table = dis_table[rownames(design2),rownames(design2)]
    sub_dis_table = as.dist(sub_dis_table, diag = FALSE, upper = FALSE)
    adonis_table = adonis(sub_dis_table ~ group, data = design2, permutations = 1000) 
    adonis_pvalue = adonis_table$aov.tab$`Pr(>F)`[1]
    print(paste("In", m, "pvalue between", sampleA, "and", sampleB, "is", adonis_pvalue, sep=" "))
    adonis_pvalue = paste(m, sampleA, sampleB, adonis_pvalue, sep="\t")
    return(adonis_pvalue)
  }
}

design = read.table("../data/design.txt", header=T, row.names=1, sep="\t")
design$group=design$groupID

if (TRUE){
  sub_design = subset(design, group %in% c("DiseaseRootspike","HealthyRootspike","BulksoilBulksoilspike"))
  sub_design$group  = factor(sub_design$group, levels=c("DiseaseRootspike","HealthyRootspike","BulksoilBulksoilspike"))
}else{
  sub_design = design
}
# 
library(dplyr)
sub_design_t = as.data.frame(t(sub_design))
sub_design_2 = dplyr::select(sub_design_t, TABS1, TABS2, TABS3, TABS4, TABS5, TAD2, TAD3, TAD4, TAD6, TAD7, TAD9, TAD10, TAH1, TAH3, TAH4, TAH5, TAH6, TAH8, TAH10)
sub_design = as.data.frame(t(sub_design_2))
```


```{r}
beta = read.table(paste("../data/beta/bray_curtis.txt",sep=""), header=T, row.names=1, sep="\t", comment.char="") 
idx = rownames(sub_design) %in% rownames(beta)
sub_design=sub_design[idx,]
sub_beta=beta[rownames(sub_design),rownames(sub_design)]

pcoa = cmdscale(sub_beta, k=4, eig=T) # k is dimension, 3 is recommended; eig is eigenvalues
points = as.data.frame(pcoa$points) # get coordinate string, format to dataframme
eig = pcoa$eig
points = cbind(points, sub_design$group)
colnames(points) = c("PC1", "PC2", "PC3", "PC4","group") 

p = ggplot(points, aes(x=PC1, y=PC2, color=group, shape= group)) + geom_point(alpha=.7, size=2) +
    labs(x=paste("PCoA 1 (", format(100 * eig[1] / sum(eig), digits=4), "%)", sep=""),
         y=paste("PCoA 2 (", format(100 * eig[2] / sum(eig), digits=4), "%)", sep=""),
         title=paste("bray_curtis PCoA",sep="")) + theme_classic() + 
    scale_shape_manual(values=c(16, 17, 15))+ 
    scale_color_manual(values=c("#C80000", "#654321", "#000000"))+ 
    theme(legend.position="top")
p
```


## Fig 6b

```{r}

design = read.table("../data/design.txt", header=T, row.names=1, sep="\t")
design$group=design$groupID

if (TRUE){
  sub_design = subset(design, group %in% c("DiseaseRootspike","HealthyRootspike","BulksoilBulksoilspike"))
  sub_design$group  = factor(sub_design$group, levels=c("DiseaseRootspike","HealthyRootspike","BulksoilBulksoilspike"))
}else{
  sub_design = design
}
# 
library(dplyr)
sub_design_t = as.data.frame(t(sub_design))
sub_design_2 = dplyr::select(sub_design_t, TABS1, TABS2, TABS3, TABS4, TABS5, TAD2, TAD3, TAD4, TAD6, TAD7, TAD9, TAD10, TAH1, TAH3, TAH4, TAH5, TAH6, TAH8, TAH10)
sub_design = as.data.frame(t(sub_design_2))
```


```{r}
beta = read.table(paste("../data/beta/fungi_bray_curtis.txt",sep=""), header=T, row.names=1, sep="\t", comment.char="") 
idx = rownames(sub_design) %in% rownames(beta)
sub_design=sub_design[idx,]
sub_beta=beta[rownames(sub_design),rownames(sub_design)]

pcoa = cmdscale(sub_beta, k=4, eig=T) # k is dimension, 3 is recommended; eig is eigenvalues
points = as.data.frame(pcoa$points) # get coordinate string, format to dataframme
eig = pcoa$eig
points = cbind(points, sub_design$group)
colnames(points) = c("PC1", "PC2", "PC3", "PC4","group") 

p = ggplot(points, aes(x=PC1, y=PC2, color=group, shape= group)) + geom_point(alpha=.7, size=2) +
    labs(x=paste("PCoA 1 (", format(100 * eig[1] / sum(eig), digits=4), "%)", sep=""),
         y=paste("PCoA 2 (", format(100 * eig[2] / sum(eig), digits=4), "%)", sep=""),
         title=paste("bray_curtis PCoA",sep="")) + theme_classic() + 
    scale_shape_manual(values=c(16, 17, 15))+ 
    scale_color_manual(values=c("#C80000", "#654321", "#000000"))+ 
    theme(legend.position="top")
p
```




## Fig 6c

```{r}
data = read.table("../data/cluster_total_load.txt",header = T, row.names = 1)
head(data)

sub_data = subset(data, SampleID %in% c("TAD2", "TAD3", "TAD4", "TAD6", "TAD7", "TAD9", "TAD10", "TAH1", "TAH3", "TAH4", "TAH5", "TAH6", "TAH8", "TAH10"))
```


```{r}
com_w1 <- compare_means( Totalload ~ Treatment, data = sub_data, method = "wilcox.test", paired=FALSE, group.by = "Micro")
com_t1 <- compare_means( Totalload ~ Treatment, data = sub_data, method = "t.test", paired=FALSE, group.by = "Micro")
head(com_t1)
head(com_w1)

com <- rbind(com_w1, com_t1)
# write.table(com , file = "./ttload_cluster_com.txt",sep = '\t',row.names = T)
```


```{r}
p1 = ggbarplot(sub_data, x = "Treatment", y = "Totalload", add = "mean_se", facet.by = "Micro",ylim = c(0, 2500))+
  stat_compare_means(method = "t.test", label =  "p.signif", label.y = 2500, label.x = 1.5)  
p1
# ggsave(paste("./ttload_cluster_com.pdf", sep=""), p1, width = 6, height = 8)
# ggsave(paste("./ttload_cluster_com.png", sep=""), p1, width = 6, height = 8)

```

## Fig 6d

```{r}
# Public file 1. "design.txt"  Design of experiment
design = read.table("../data/design.txt", header=T, row.names= 1, sep="\t") 

# Public file 2. "otu_table.txt"  raw reads count of each OTU in each sample
otu_table = read.delim("../data/otu_plant_aa.txt", row.names= 1,  header=T, sep="\t")

# Public file 3. "rep_seqs_tax.txt"  taxonomy for each OTU, tab seperated
taxonomy = read.delim("../data/taxonomy_8.txt", row.names= 1,header=F, sep="\t")
colnames(taxonomy) = c("kingdom","phylum","class","order","family","genus","species")

tax_count = merge(taxonomy, otu_table, by="row.names")

```

```{r}
## remove the abnormal points
sub_tax_count = dplyr::select(tax_count, Row.names, kingdom, phylum, class, order, family, genus, species, TAD2, TAD3, TAD4, TAD6, TAD7, TAD9, TAD10, TAH1, TAH3, TAH4, TAH5, TAH6, TAH8, TAH10)



g1_Bipolaris = dplyr::filter(sub_tax_count, genus == "Bipolaris")
g2_Fusarium = dplyr::filter(sub_tax_count, genus == "Fusarium")
genus = rbind(g1_Bipolaris, g2_Fusarium)
otuid = as.data.frame(genus$Row.names)
colnames(otuid) = "Row.names"


com_genus = merge(sub_tax_count, otuid, by="Row.names")
colnames(com_genus)[1] = "OTUID"


com_genus_tidy = gather(com_genus, key = SampleID , value = method2aa, `TAD2`:`TAH10`)


design2 = read.table("../data/design.txt", header=T, sep="\t") 
com_genus_tidy_design = merge(com_genus_tidy, design2, by="SampleID")
com_aa = dplyr::select(com_genus_tidy_design, SampleID, OTUID, genus, method2aa, Genotype)
colnames(com_aa)[3] = "Genus"
colnames(com_aa)[5] = "Treatment"
# 
# write.table(com_aa, file = "./pointD1D5D8H2H7H9/com_aa.txt",sep = '\t',row.names = T)
```



```{r}
## 读取com_ra数据
com_ra <-  read.table("../data/com_ra.txt", header=T, sep="\t") 
com_method1ra = gather(com_ra, key = Method , value = value, `method1ra`)
com_method2aa = gather(com_aa, key = Method , value = value, `method2aa`)

## 整合compare总表
com_aara <- rbind(com_method1ra, com_method2aa)
```



```{r}
## 使用两种方法检测OTU在不同样品中的丰度
OTU45 <- dplyr::filter(com_aara, OTUID == "OTU_45")

p45 <- ggboxplot(OTU45, x="Treatment",y = "value",
               facet.by = "Method", color = "Treatment", order = c("Disease","Healthy"),
               palette =  "uchicago", add = "jitter", short.panel.labs = FALSE, title = "OTU_45_Bipolaris")
# 添加p值
p045 <- p45 + stat_compare_means(label =  "p.signif",  method = "wilcox.test",label.x = 1.55)

p045

# ggsave(paste("./pointD1D5D8H2H7H9/com_aa_OTU_45.pdf", sep=""), p032, width = 6, height = 7)
# ggsave(paste("./pointD1D5D8H2H7H9/com_aa_OTU_45.png", sep=""), p032, width = 6, height = 7)
```


## Fig 6e

```{r, warning=FALSE, message=FALSE}
data = read.table("../data/Fig6g.txt",header = T, row.names = 1)
head(data)

data_1 <- filter(data, re == "1")
data_2 <- filter(data, re == "2")
```

```{r, warning=FALSE, message=FALSE}
com_1 <- compare_means( ratio ~ treatment, data = data_1, method = "wilcox.test", paired=FALSE)
com_t1 <- compare_means( ratio ~ treatment, data = data_1, method = "t.test", paired=FALSE)
com_2 <- compare_means( ratio ~ treatment, data = data_2, method = "wilcox.test", paired=FALSE)
com_t2 <- compare_means( ratio ~ treatment, data = data_2, method = "t.test", paired=FALSE)
head(com_1)
head(com_t1)
head(com_2)
head(com_t2)

com <- rbind(com_1, com_t1, com_2, com_t2)

# 
# write.table(com , file = "./result/Fig6_load_com.txt",sep = '\t',row.names = T)
```

```{r}
p1 = ggbarplot(data_1, x = "treatment", y = "ratio", add = "mean_se")+
  stat_compare_means(method = "t.test")  
p1
# ggsave(paste("./result/Fig6-load-1.pdf", sep=""), p1, width = 3, height = 6)
```


## Fig 6f

```{r}
rm(list=ls())
options(warn=3)

source("plot_function.R")
library(ggcorrplot)
library(corrplot)
library(pheatmap)
library(ggpubr)
library(vegan)
library(ggplot2)
library(ggsignif)
library("Biobase", quietly=T, warn.conflicts=F)
library("RColorBrewer", quietly=T, warn.conflicts=F)
library("reshape2", quietly=T, warn.conflicts=F)
library(dplyr)
library(tidyr)
library(psych)

## to set the date as the version
########flag==TRUE,usearch ; FALSE, unoise; project_name:.project; .version: as output version

.flag <- TRUE
# .flag <- FALSE
## to set the date as the version
.date <- Sys.Date()

### sub version swich
.subflag <- TRUE
# .subv <- FALSE

if(.subflag == TRUE){
  .sub_ver <- "Top300"
  threashold=300
}else
{
  .sub_ver <- ''
}


if(.flag==TRUE)
{.meth <- "Usearch" 
}else
{.meth <- "Unoise"}


.project <- "ConetRootDis"
.version <- paste0(.date,.project,.sub_ver,.meth)

```


```{r,warning=FALSE, message=FALSE}
######################## AA network construction removal of sample TAD1,TAD5 TAD8 TAH2 TAH7 TAH9

design = read.table("../data/doc/design.txt", header=T, row.names= 1, sep="\t") 
sub_design <- design[design$Spikein %in% "spike"& !design$Other %in% "Bulksoil",]
id_rm= c("TAD1","TAD5","TAD8","TAH2","TAH7","TAH9")
sub_design= sub_design[!rownames(sub_design) %in% id_rm,]


bac_aa <- read.delim("../data/Usearch/AA/otutab_norm_AA.txt", row.names= 1,header=T, sep="\t",stringsAsFactors = F)
fungi_aa <- read.delim("../data/Usearch/fAA/otutab_norm_AA.txt", row.names= 1,header=T, sep="\t",stringsAsFactors = F)

id <- colnames(fungi_aa)%in%colnames(bac_aa)

fungi_aa <- fungi_aa[,id]

id <- match(colnames(bac_aa),colnames(fungi_aa))
bac_aa <- bac_aa[,id]
bac_aa <- bac_aa[,colnames(bac_aa)%in% rownames(sub_design)]
fungi_aa <- fungi_aa[,colnames(fungi_aa)%in% rownames(sub_design)]


bac_aa$species <- rep("bac",nrow(bac_aa))
fungi_aa$species <- rep("fungi",nrow(fungi_aa))
merge_bac_fungi <-rbind(bac_aa,fungi_aa) 

###1. Filtering OTUs by occurrence frequency (i.e.,number of samples an OTU is Present 60% of the samples)
Abu <- merge_bac_fungi
Abu$species <- NULL
table <- Abu
table[table>0.0001] <- 1
table.generalist <- Abu[which(rowSums(table)>=5),]  ### 5 out of 14
Abu <- table.generalist

merge_bac_fungi <- merge_bac_fungi[rownames(merge_bac_fungi) %in% rownames(Abu),]



## Tax ##
bac_tax <- read.delim("../data/Usearch/Tax/Tax.txt", row.names= 1,header=T, sep="\t",stringsAsFactors = F)
fungi_tax <- read.delim("../data/Usearch/fTax/fTax.txt", row.names= 1,header=F, sep="\t",stringsAsFactors = F)

colnames(fungi_tax)<- colnames(bac_tax)
# fungi_tax$V10 <- NULL
# fungi_tax$V9 <- NULL

colnames(fungi_tax) <- colnames(bac_tax)

tax_merge_bac_fungi <-rbind(fungi_tax,bac_tax) 
tax_merge_bac_fungi <- tax_merge_bac_fungi[-1,]

tax_merge_bac_fungi$ID <- rownames(tax_merge_bac_fungi)
merge_bac_fungi$ID <- rownames(merge_bac_fungi)
merge_abun_tax<- merge(merge_bac_fungi,tax_merge_bac_fungi,by = "ID")
```

```{r,warning=FALSE, message=FALSE}
######dplyr to data manipulation
# install.packages("psych")
# library(psych)

aa_dis_a<- merge_abun_tax[,c(2:15,17,22)] %>%
  mutate(rowSumAbun=rowSums(merge_abun_tax[,c(2:15)])) %>%
  group_by(Kingdom,Genus) %>%
  summarize_all(sum) %>%
  arrange(desc(rowSumAbun))

# colnames(aa_dis_a)

aa_dis_a<- as.data.frame(aa_dis_a)
rownames(aa_dis_a)<- paste(aa_dis_a$Kingdom,aa_dis_a$Genus,sep = '_')
aa_dis_a$Kingdom <- NULL
aa_dis_a$Genus <- NULL
aa_dis_a <- as.data.frame(t(aa_dis_a))
aa_dis_a$Fungi_Unassigned <- NULL


#### correlation of top 50 genus with microbial load
dis_micload<- as.data.frame(rowSums(aa_dis_a))
colnames(dis_micload) <- "microbial load"

top50_cor_micload_pearson<- corr.test(aa_dis_a[,c(1:50)],dis_micload,use = "complete",method="pearson",adjust = "fdr",alpha = .05)
# top50_cor_micload_pearson$p
# top50_cor_micload_pearson$r

top50_cor_micload_spearman<- corr.test(aa_dis_a[,c(1:ncol(aa_dis_a))],dis_micload,use = "complete",method="spearman",adjust = "fdr",alpha = .05)
# top50_cor_micload_spearman$p
# top50_cor_micload_spearman$r


idd <- grep("Bipolaris",colnames(aa_dis_a)) 
idd1 <- grep("Fusarium",colnames(aa_dis_a)) 

top50_cor_micload_pearson<- corr.test(aa_dis_a[,c(idd)],dis_micload,use = "complete",method="pearson",adjust = "fdr",alpha = .05)
# top50_cor_micload_pearson$p
# top50_cor_micload_pearson$r


top50_cor_micload_spearman<- corr.test(aa_dis_a[,c(idd)],dis_micload,use = "complete",method="spearman",adjust = "fdr",alpha = .05)
# top50_cor_micload_spearman$p
# top50_cor_micload_spearman$r


top50_cor_micload_pearson<- corr.test(aa_dis_a[,c(idd1)],dis_micload,use = "complete",method="pearson",adjust = "fdr",alpha = .05)
# top50_cor_micload_pearson$p
# top50_cor_micload_pearson$r


top50_cor_micload_spearman<- corr.test(aa_dis_a[,c(idd1)],dis_micload,use = "complete",method="spearman",adjust = "fdr",alpha = .05)
# top50_cor_micload_spearman$p
# top50_cor_micload_spearman$r


#### note the most 210 threshold
if (.subflag==TRUE){
  col_length=threashold
}else{
  col_length <- ncol(aa_dis_a)
}

aa_dis_occor_spearman <- corr.test(aa_dis_a[1:col_length],use="pairwise",method="spearman",adjust="fdr",alpha=.05)
# aa_dis_occor_spearman$r

# ## to filter the top 30 genus according to AA profiling
# rownames(aa_dis_occor_spearman$r)

idd <- grep("Fungi_Bipolaris",rownames(aa_dis_occor_spearman$r))

idd1 <- grep("Fungi_Fusarium",rownames(aa_dis_occor_spearman$r))

a_dis=aa_dis_occor_spearman$r
a_dis_p=aa_dis_occor_spearman$p
rr1 <- a_dis_p
rr1[lower.tri(rr1)] <- t(rr1)[lower.tri(rr1)]
a_dis_p<- rr1



if(.flag == TRUE){
  
  aa_dis_sper_mat_top30 <- aa_dis_occor_spearman$r[c(1:55,idd),c(1:55,idd)]
  # aa_dis_sper_p_top30 <- aa_dis_occor_spearman$p[c(1:55,idd),c(1:55,idd)]
  aa_dis_sper_p_top30 <- aa_dis_occor_spearman$p[c(1:55,idd),c(1:55,idd)]
  # colnames(aa_dis_sper_mat_top30)
  
  name_vector <- c("Bacteria_Promicromonospora","Bacteria_Pseudomonas","Bacteria_Phyllobacterium","Bacteria_Variovorax",
                   "Bacteria_Flavobacterium","Bacteria_Rhizobium","Bacteria_Lentzea","Bacteria_Pseudoxanthomonas",
                   "Bacteria_Sphingomonas","Bacteria_Nocardioides","Bacteria_Actinoplanes","Bacteria_Streptomyces",
                   "Bacteria_Chryseobacterium","Bacteria_Herbaspirillum","Bacteria_Massilia",
                   "Bacteria_Devosia","Bacteria_Kribbella","Bacteria_Mycobacterium",
                   "Bacteria_Microbacterium","Bacteria_Mesorhizobium","Bacteria_Stenotrophomonas","Bacteria_Sphingopyxis",
                   "Bacteria_Aeromicrobium","Bacteria_Caulobacter","Bacteria_Steroidobacter","Bacteria_Arthrobacter",
                   "Bacteria_Umezawaea","Bacteria_Chondromyces","Bacteria_Sorangium","Bacteria_Agromyces",
                   "Bacteria_Rhodanobacter","Bacteria_Galbitalea","Bacteria_Pantoea","Bacteria_Shinella",
                   "Bacteria_Bosea","Bacteria_Solirubrobacter","Bacteria_Cellulomonas",
                   "Bacteria_Lysobacter","Bacteria_Bradyrhizobium","Bacteria_Rudaibacter","Bacteria_Glycomyces",
                   "Bacteria_Burkholderia","Bacteria_Altererythrobacter","Bacteria_Bacillus","Bacteria_Amycolatopsis",
                   "Bacteria_Rhodopseudomonas","Bacteria_Phycicoccus","Bacteria_Sphingobium","Bacteria_Gaiella",
                   "Bacteria_Blastococcus","Bacteria_Ideonella",
                   "Fungi_Acremonium","Fungi_Scytalidium","Fungi_Gibberella","Fungi_Holtermanniella" ,
                   
                   "Fungi_Bipolaris" )
  
  ids <- match(name_vector,row.names(aa_dis_sper_mat_top30))

  aa_dis_sper_mat_top30<- aa_dis_sper_mat_top30[ids,ids]
  aa_dis_sper_p_top30 <- aa_dis_sper_p_top30[ids,ids]
   corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .05,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  dev.off()
  
  # pdf(file ="../plot/rootDisease/190318_usearch_ordername_shuffle_AA_sper_mat_top50_order_upper_p_value_0.01.pdf" )
  # pdf(file = paste0(figures.dir,.version,"OrdShfAASperMatTop55FuBiUppTriP0.01.pdf") )
  corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .01,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  dev.off()
  
  
  corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .001,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  #dev.off()
  
}else
{
  aa_dis_sper_mat_top30 <- aa_dis_occor_spearman$r[c(1:55,idd),c(1:55,idd)]
  # aa_dis_sper_p_top30 <- aa_dis_occor_spearman$p[c(1:55,idd),c(1:55,idd)]
  aa_dis_sper_p_top30 <- aa_dis_occor_spearman$p[c(1:55,idd),c(1:55,idd)]
  
  fh <- paste0(table.dir,.version,"AssociationGenusNameSort.txt")
  sink(fh)
  # colnames(aa_dis_sper_mat_top30)
  # sort(colnames(aa_dis_sper_mat_top30))
  sink()
  
  system(paste0("sed -i 's/\\s\\+/,/g' ",fh))
  system(paste0("sed -i 's/^,//g' ",fh))
  system(paste0("sed -i 's/^\\[[1-9]\\+\\],//g' ",fh))
  
  
  name_vector <- c("Bacteria_Promicromonospora","Bacteria_Pseudomonas","Bacteria_Phyllobacterium","Bacteria_Rhizobium","Bacteria_Flavobacterium",
                   "Bacteria_Variovorax","Bacteria_Lentzea","Bacteria_Sphingomonas","Bacteria_Nocardioides","Bacteria_Streptomyces",
                   "Bacteria_Chryseobacterium","Bacteria_Actinoplanes","Bacteria_Pseudoxanthomonas",
                   "Bacteria_Devosia","Bacteria_Kribbella","Bacteria_Mycobacterium","Bacteria_Lysobacter","Bacteria_Microbacterium",
                   "Bacteria_Massilia","Bacteria_Herbaspirillum","Bacteria_Mesorhizobium","Bacteria_Stenotrophomonas","Bacteria_Aeromicrobium",
                   "Bacteria_Sphingopyxis","Bacteria_Steroidobacter","Bacteria_Caulobacter","Bacteria_Arthrobacter","Bacteria_Umezawaea",
                   "Bacteria_Agromyces","Bacteria_Shinella","Bacteria_Chondromyces","Bacteria_Sorangium","Bacteria_Pantoea",
                   "Bacteria_Solirubrobacter","Bacteria_Bradyrhizobium","Bacteria_Bosea","Bacteria_Couchioplanes","Bacteria_Cellulomonas",
                   "Bacteria_Polaromonas","Bacteria_Galbitalea","Bacteria_Burkholderia","Bacteria_Rhodanobacter",
                   "Bacteria_Glycomyces","Bacteria_Xanthomonas","Bacteria_Bacillus",
                   "Bacteria_Rudaibacter","Bacteria_Rhodopseudomonas","Bacteria_Altererythrobacter","Bacteria_Pseudonocardia","Bacteria_Phycicoccus",
                   "Fungi_Acremonium","Fungi_Scytalidium","Fungi_Podospora","Fungi_Myrmecridium","Fungi_Fusarium",
                   "Fungi_Bipolaris" )
  
  ids <- match(name_vector,row.names(aa_dis_sper_mat_top30))
  # 
  aa_dis_sper_mat_top30<- aa_dis_sper_mat_top30[ids,ids]
  aa_dis_sper_p_top30 <- aa_dis_sper_p_top30[ids,ids]
  
  corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .05,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  dev.off()
  
  corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .01,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  dev.off()
  
  
  corrplot(aa_dis_sper_mat_top30,p.mat = aa_dis_sper_p_top30, sig.level = .001,
           insig = "blank", addrect = 2,type = "upper",diag = F,
           tl.col = "black",tl.cex = 0.75,
           cl.pos = "b",cl.length = 11,cl.align.text = "l",
           cl.lim = c(-1,1),
           title = "Genus-genus spearman correlation")
  
  #dev.off()
  
}


#### correlation of disease-associated genus with microbial load
dis_micload<- as.data.frame(rowSums(aa_dis_a))
colnames(dis_micload) <- "microbial load"


top50_cor_micload_pearson<- corr.test(aa_dis_a[,ids],dis_micload,use = "complete",method="pearson",adjust = "fdr",alpha = .05)
# top50_cor_micload_pearson$p
# top50_cor_micload_pearson$r

top50_cor_micload_spearman<- corr.test(aa_dis_a[,ids],dis_micload,use = "complete",method="spearman",adjust = "fdr",alpha = .05)
# top50_cor_micload_spearman$p
# top50_cor_micload_spearman$r
```

```{r,warning=FALSE, message=FALSE}
### RA network construction
design = read.table("../data/doc/design.txt", header=T, row.names= 1, sep="\t") 

sub_design <- design[design$Spikein %in% "spike"& !design$Other %in% "Bulksoil",]
id_rm= c("TAD1","TAD5","TAD8","TAH2","TAH7","TAH9")
sub_design= sub_design[!rownames(sub_design) %in% id_rm,]

bac_aa <- read.delim("../data/Usearch/RA/otutab_norm_RA.txt",  row.names= 1,header=T, sep="\t",stringsAsFactors = F)
fungi_aa <- read.delim("../data/Usearch/fRA/otutab_norm_RA.txt", row.names= 1,header=T, sep="\t",stringsAsFactors = F)

id <- colnames(fungi_aa) %in% colnames(bac_aa)
fungi_aa <- fungi_aa[,id]

id <- match(colnames(bac_aa),colnames(fungi_aa))
bac_aa <- bac_aa[,id]
bac_aa <- bac_aa[,colnames(bac_aa)%in% rownames(sub_design)]
fungi_aa <- fungi_aa[,colnames(fungi_aa)%in% rownames(sub_design)]





bac_aa$species <- rep("bac",nrow(bac_aa))
fungi_aa$species <- rep("fungi",nrow(fungi_aa))
merge_bac_fungi <-rbind(bac_aa,fungi_aa) 

colnames(fungi_tax) <- colnames(bac_tax)

tax_merge_bac_fungi <-rbind(fungi_tax,bac_tax) 
tax_merge_bac_fungi <- tax_merge_bac_fungi[-1,]
tax_merge_bac_fungi$ID <- rownames(tax_merge_bac_fungi)
merge_bac_fungi$ID <- rownames(merge_bac_fungi)
merge_abun_tax<- merge(merge_bac_fungi,tax_merge_bac_fungi,by = "ID")

######dplyr to data manipulation
ra_dis_a<- merge_abun_tax[,c(2:15,17,22)] %>%
  mutate(rowSumAbun=rowSums(merge_abun_tax[,c(2:15)])) %>%
  group_by(Kingdom,Genus) %>%
  summarize_all(sum) %>%
  arrange(desc(rowSumAbun))

ra_dis_a<- as.data.frame(ra_dis_a)
rownames(ra_dis_a)<- paste(ra_dis_a$Kingdom,ra_dis_a$Genus,sep = '_')
ra_dis_a$Kingdom <- NULL
ra_dis_a$Genus <- NULL
ra_dis_a <- as.data.frame(t(ra_dis_a))
ra_dis_a$Fungi_Unassigned <- NULL
ra_dis_a<- ra_dis_a[,colnames(ra_dis_a) %in% colnames(aa_dis_a)]


if (.subflag==TRUE){
  col_length <- threashold
  
}else
{
  col_length <- 350
}

ra_dis_occor_spearman <- corr.test(ra_dis_a[1:col_length],use="pairwise",method="spearman",adjust="fdr",alpha=.05)
# ra_dis_occor_spearman$r

## to filter the top 30 genus according to AA profiling
id <- rownames(ra_dis_occor_spearman$r)%in% rownames(aa_dis_sper_p_top30)
ra_dis_sper_mat_top30 <- ra_dis_occor_spearman$r[id,id]
ra_dis_sper_p_top30 <- ra_dis_occor_spearman$p[id,id]
ids <- match(row.names(aa_dis_sper_mat_top30),row.names(ra_dis_sper_mat_top30))

ra_dis_sper_mat_top30<- ra_dis_sper_mat_top30[ids,ids]
ra_dis_sper_p_top30 <- ra_dis_sper_p_top30[ids,ids]
top50_plus_other <- ra_dis_a[,colnames(ra_dis_a)%in% rownames(ra_dis_sper_mat_top30)]

corrplot(ra_dis_sper_mat_top30,p.mat = ra_dis_sper_p_top30, sig.level = .05,
         insig = "blank", addrect = 2,type = "upper",diag = F,col = cm.colors(100),
         tl.col = "black",tl.cex = 0.75,
         cl.pos = "b",cl.length = 11,cl.align.text = "l",
         cl.lim = c(-1,1),
         title = "Genus-genus spearman correlation")

dev.off()

corrplot(ra_dis_sper_mat_top30,p.mat = ra_dis_sper_p_top30, sig.level = .01,
         insig = "blank", addrect = 2,type = "upper",diag = F,col = cm.colors(100),
         tl.col = "black",tl.cex = 0.75,
         cl.pos = "b",cl.length = 11,cl.align.text = "l",
         cl.lim = c(-1,1),
         title = "Genus-genus spearman correlation")

dev.off()

corrplot(ra_dis_sper_mat_top30,p.mat = ra_dis_sper_p_top30, sig.level = .001,
         insig = "blank", addrect = 2,type = "upper",diag = F,col = cm.colors(100),
         tl.col = "black",tl.cex = 0.75,
         cl.pos = "b",cl.length = 11,cl.align.text = "l",
         cl.lim = c(-1,1),
         title = "Genus-genus spearman correlation")

#dev.off()


```


